### PR TITLE
HW 11

### PR DESCRIPTION
**Pull request by geo-tat**

### CODE REVIEW

File: src/main/java/com/andersen/hw/service/TicketServiceImpl.java [hunk 1]
```diff
@@ -56,13 +67,24 @@ public void deleteById(Integer id) {
...
 
+    @Override
+    @Transactional
+    public void updateUserStatusAndCreateTicket(User user, Ticket ticket) {
+        if (updateUserFlag) {
+            userStorageDao.updateUserStatus(user);
+            ticketStorageDao.addTicket(ticket);
+        }
+        throw new IllegalArgumentException("The operation is disabled now");
+    }
```
Comment (AtticSiol) [inline src/main/java/com/andersen/hw/service/TicketServiceImpl.java]: in this implementation exception will be thrown every time.  revert the logic:
```java
    public void updateUserStatusAndCreateTicket(User user, Ticket ticket) {
        if (!updateUserFlag) {
            throw new IllegalArgumentException("The operation is disabled now");
        }
        userStorageDao.updateUserStatus(user);
        ticketStorageDao.addTicket(ticket);
    }
```

File: src/main/java/com/andersen/hw/storage/UserStorageDao.java [hunk 2]
```diff
@@ -1,119 +1,60 @@
...
 
 @Repository
 public class UserStorageDao implements UserStorage {
-    private final DataSource dataSource;
+    private final JdbcTemplate jdbcTemplate;
+    private final UserMapper userMapper;
     private final int classId;
 
-    public UserStorageDao(DataSource dataSource) {
+    @Autowired
```
Comment (AtticSiol) [inline src/main/java/com/andersen/hw/storage/UserStorageDao.java]: this annotation can be skipped when you have just 1 constructor into the class

File: src/main/java/com/andersen/hw/service/TicketServiceImpl.java (hunk 1 above)
Comment (geo-tat) [reply src/main/java/com/andersen/hw/service/TicketServiceImpl.java]: done

File: src/main/java/com/andersen/hw/storage/UserStorageDao.java (hunk 2 above)
Comment (geo-tat) [reply src/main/java/com/andersen/hw/storage/UserStorageDao.java]: done

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**